src/go/rpk/pkg/config/config.go

```diff
@@ -235,33 +240,59 @@ func checkRedpandaConfig(v *viper.Viper) []error {
 		errs = append(errs, fmt.Errorf("redpanda.node_id can't be a negative integer"))
 	}
 
-	apiConfigKeys := []string{"redpanda.rpc_server", "redpanda.kafka_api"}
-
-	for _, key := range apiConfigKeys {
-		exists := v.Sub(key) != nil
-		if !exists {
+	rpcServerKey := "redpanda.rpc_server"
+	exists := v.Sub(rpcServerKey) != nil
+	if !exists {
+		errs = append(
+			errs,
+			fmt.Errorf("%s missing", rpcServerKey),
+		)
+	} else {
+		socket := &SocketAddress{}
+		err := v.UnmarshalKey(rpcServerKey, socket)
+		if err != nil {
 			errs = append(
 				errs,
-				fmt.Errorf("%s missing", key),
+				fmt.Errorf("invalid structure for %s", rpcServerKey),
 			)
 		} else {
-			socket := &SocketAddress{}
-			err := v.UnmarshalKey(key, socket)
-			if err != nil {
-				errs = append(
-					errs,
-					fmt.Errorf("invalid structure for %s", key),
-				)
-			} else {
-				errs = append(
-					errs,
-					checkSocketAddress(*socket, key)...,
-				)
-			}
+			errs = append(
+				errs,
+				checkSocketAddress(*socket, rpcServerKey)...,
+			)
+		}
+	}
+
+	var kafkaListeners []NamedSocketAddress
+	err := v.UnmarshalKey("redpanda.kafka_api", &kafkaListeners)
```

Comment:
In the `redpanda.rpc_server` there is check (line 244-245) if the key exist in viper configuration. Should we have the same check for `redpanda.kafka_api`?

Reply:
You're right, we should.